File: src/main/java/legendarena/api/utils/Cooldown.java
package legendarena.api.utils;

import org.bukkit.ChatColor;

/**
 * Code stolen from: http://hastebin.com/noqemohuda.java with various changes
 */
public class Cooldown {

    private double seconds;
    private long start;

    public Cooldown(double seconds) {
        this.seconds = seconds;
        this.start = System.currentTimeMillis();
    }

    public boolean done() {
        return ((System.currentTimeMillis() - start) > seconds * 1000);
    }

    public String remainingSeconds() {
        return String.format("%.1f", seconds - ((System.currentTimeMillis() - start) / 1000.0));
    }

    public String getTimeRemaining() {
        return ChatUtils.Messages.getCustomMsg("Cooldown") + remainingSeconds() + ChatColor.GRAY + "/" + ChatColor.BLUE + seconds + "s";
    }

}

File: src/main/java/legendarena/api/utils/Cooldown.kt
package legendarena.api.utils

import org.bukkit.ChatColor

class Cooldown {

    private val seconds: Double
    private val start: Long

    constructor(seconds: Double) {
        this.seconds = seconds
        this.start = System.currentTimeMillis()
    }

    fun done(): Boolean {
        if(java.lang.String.format("%.1f", seconds - ((System.currentTimeMillis() - start) / 1000.0)) == "0.0") return true //fucking idiotic cooldown system bugs
        return ((System.currentTimeMillis() - start) > seconds * 1000)
    }

    fun remainingSeconds(): String {
        return java.lang.String.format("%.1f", seconds - ((System.currentTimeMillis() - start) / 1000.0))
    }

    fun getTimeRemaining(): String {
        //return ChatUtils.getCustomMsg("Cooldown") + remainingSeconds() + ChatColor.GRAY + "/" + ChatColor.BLUE + seconds + "s"
        return ChatUtils.getFormattedMsg("Remaining cooldown", remainingSeconds() + ChatColor.GRAY + "/" + ChatColor.BLUE + seconds + "s")
    }

}